axelrod/result_set.py

```diff
@@ -0,0 +1,54 @@
+import numpy
+from StringIO import StringIO
+
+
+class ResultSet(object):
+    """
+    A class to hold the results of a tournament
+    and to return them as plots or csv files.
+    """
+
+    def __init__(self, players, turns, repetitions):
+        self.players = players
+        self.nplayers = len(players)
+        player_list = list(range(len(players)))
+        self.turns = turns
+        self.repetitions = repetitions
+        repetition_list = list(range(repetitions))
+        self.results = [
+            [[0 for irep in repetition_list]
+                for j in player_list] for i in player_list]
+        self.output_initialised = False
+
+    def generate_scores(self):
+        numpy_array_results = numpy.array(self.results)
+        return numpy_array_results.sum(axis=1)
+
+    def generate_ranking(self, scores):
+        ranking = sorted(
+            range(self.nplayers),
+            key=lambda i: numpy.median(scores[i]))
+        return ranking
+
+    def generate_ranked_names(self, ranking):
```

Comment:
Same comment here (and throughout if possible). I'm going to need to find that 8th and 10th day in the week and at some point go through everything and add at least one line of docstring but I can at least start asking for it from now...